diff --git a/src/password-store.sh b/src/password-store.sh
index a0dcf2e..1c9a6cb 100755
--- a/src/password-store.sh
+++ b/src/password-store.sh
@@ -536,8 +536,8 @@ cmd_generate() {
 
 	[[ $inplace -eq 0 && $force -eq 0 && -e $passfile ]] && yesno "An entry already exists for $path. Overwrite it?"
 
-	read -r -n $length pass < <(LC_ALL=C tr -dc "$characters" < /dev/urandom)
-	[[ ${#pass} -eq $length ]] || die "Could not generate password from /dev/urandom."
+	read -r -n $length pass < <(dd iflag=fullblock if=/dev/random | LC_ALL=C tr -dc "$characters")
+	[[ ${#pass} -eq $length ]] || die "Could not generate password from /dev/random."
 	if [[ $inplace -eq 0 ]]; then
 		echo "$pass" | $GPG -e "${GPG_RECIPIENT_ARGS[@]}" -o "$passfile" "${GPG_OPTS[@]}" || die "Password encryption aborted."
 	else
