diff --git a/src/password-store.sh b/src/password-store.sh
index 567fd55ca357142cdf4d83dc044dc58a6964b746..b6c9b5a9114d202f978227bca307c5785d34f665 100755
--- a/src/password-store.sh
+++ b/src/password-store.sh
@@ -718,4 +718,8 @@ case "$1" in
 	git) shift;			cmd_git "$@" ;;
 	*)				cmd_extension_or_show "$@" ;;
 esac
+
+# Make GnuPG always ask for the smartcard pin
+gpgconf --reload scdaemon
+
 exit 0
